internal/gengapic/gengapic.go

```diff
@@ -52,82 +51,164 @@ const (
-			return &g.resp, nil
+			return &options{sampleOnly: true}, nil
 		}
 
 		e := strings.IndexByte(s, '=')
 		if e < 0 {
-			return &g.resp, errors.E(nil, "invalid plugin option format, must be key=value: %s", s)
+			return nil, errors.E(nil, "invalid plugin option format, must be key=value: %s", s)
 		}
 
 		key, val := s[:e], s[e+1:]
 		if val == "" {
-			return &g.resp, errors.E(nil, "invalid plugin option value, missing value in key=value: %s", s)
+			return nil, errors.E(nil, "invalid plugin option value, missing value in key=value: %s", s)
 		}
 
 		switch key {
 		case "go-gapic-package":
 			p := strings.IndexByte(s, ';')
 
 			if p < 0 {
-				return &g.resp, errors.E(nil, paramError)
+				return nil, errors.E(nil, paramError)
 			}
 
-			pkgPath = s[e+1 : p]
-			pkgName = s[p+1:]
-			outDir = filepath.FromSlash(pkgPath)
+			opts.pkgPath = s[e+1 : p]
+			opts.pkgName = s[p+1:]
+			opts.outDir = filepath.FromSlash(opts.pkgPath)
 		case "gapic-service-config":
-			f, err := os.Open(val)
-			if err != nil {
-				return &g.resp, errors.E(nil, "error opening service config: %v", err)
-			}
-
-			err = yaml.NewDecoder(f).Decode(&g.serviceConfig)
-			if err != nil {
-				return &g.resp, errors.E(nil, "error decoding service config: %v", err)
-			}
+			opts.serviceConfigPath = val
 		case "grpc-service-config":
-			f, err := os.Open(val)
-			if err != nil {
-				return &g.resp, errors.E(nil, "error opening gRPC service config: %v", err)
-			}
-
-			g.grpcConf, err = conf.New(f)
-			if err != nil {
-				return &g.resp, errors.E(nil, "error parsing gPRC service config: %v", err)
-			}
+			opts.grpcConfPath = val
 		case "module":
-			g.modulePrefix = val
+			opts.modulePrefix = val
 		case "release-level":
-			g.relLvl = strings.ToLower(val)
+			opts.relLvl = strings.ToLower(val)
+		case "transport":
+			for _, t := range strings.Split(val, "+") {
+				switch t {
+				case "grpc":
+					opts.transports = append(opts.transports, grpc)
+				case "rest":
+					opts.transports = append(opts.transports, rest)
+				default:
+					return nil, errors.E(nil, "invalid transport option: %s", t)
+				}
+			}
 		}
 	}
 
-	if pkgPath == "" || pkgName == "" || outDir == "" {
-		return &g.resp, errors.E(nil, paramError)
+	if opts.pkgPath == "" || opts.pkgName == "" || opts.outDir == "" {
+		return nil, errors.E(nil, paramError)
 	}
 
-	if g.modulePrefix != "" {
-		if !strings.HasPrefix(outDir, g.modulePrefix) {
-			return &g.resp, errors.E(nil, "go-gapic-package %q does not match prefix %q", outDir, g.modulePrefix)
+	if opts.modulePrefix != "" {
+		if !strings.HasPrefix(opts.outDir, opts.modulePrefix) {
+			return nil, errors.E(nil, "go-gapic-package %q does not match prefix %q", opts.outDir, opts.modulePrefix)
 		}
-		outDir = strings.TrimPrefix(outDir, g.modulePrefix+"/")
+		opts.outDir = strings.TrimPrefix(opts.outDir, opts.modulePrefix+"/")
 	}
 
+	// Default is just grpc
+	if opts.transports == nil {
+		opts.transports = []Transport{grpc}
```

Comment:
Consider moving this assignment to line 102 to better group the initialization together.

Reply:
I think the current placement better communicates the intention that this is a default value if the user doesn't provide a `transport=` argument.